In [1]:
from sqlalchemy import create_engine

In [103]:
!pip install nbmerge

  Created wheel for nbmerge: filename=nbmerge-0.0.4-py2.py3-none-any.whl size=6413 sha256=fee222f27ac99b5891ff2085e38c27bb8f7b0cedd56ef4a627bfc9bec0207d54
  Stored in directory: c:\users\ryann\appdata\local\pip\cache\wheels\0b\a8\31\4ae563f674b488c82ee72cfe0f86487c21532db97ae910b264
Successfully built nbmerge


In [104]:
!nbmerge data_collection.ipynb sql_analysis.ipynb -o preesentation.ipynb

In [2]:
%load_ext sql

In [61]:
%sql mysql://admin:sql_2021@movie-list.crvcl4hdswes.us-east-1.rds.amazonaws.com/movie_list

# Exploratory Queries

This first query confirms that the imported data is correct and appears clean

This first query confirms that the imported data is correct and appears clean

In [80]:
%%sql
SELECT DISTINCT *
FROM movie;

 * mysql://admin:***@movie-list.crvcl4hdswes.us-east-1.rds.amazonaws.com/movie_list
250 rows affected.


[('The Batman', '\nAction, Crime, Drama            ', 'PG-13', '8.0'),
 ('The Northman', '\nAction, Adventure, Drama            ', 'R', '7.8'),
 ('Uncharted', '\nAction, Adventure            ', 'PG-13', '6.4'),
 ('Everything Everywhere All at Once', '\nAction, Adventure, Comedy            ', 'R', '8.8'),
 ('The Unbearable Weight of Massive Talent', '\nAction, Comedy, Crime            ', 'R', '7.5'),
 ('X', '\nHorror, Thriller            ', 'R', '6.7'),
 ('The Black Phone', '\nHorror            ', 'R', '7.4'),
 ('Fantastic Beasts: The Secrets of Dumbledore', '\nAdventure, Family, Fantasy            ', 'PG-13', '6.5'),
 ('Ambulance', '\nAction, Crime, Drama            ', 'R', '6.2'),
 ('Death on the Nile', '\nCrime, Drama, Mystery            ', 'PG-13', '6.3'),
 ('The Bad Guys', '\nAnimation, Adventure, Comedy            ', 'PG', '6.9'),
 ('Spider-Man: No Way Home', '\nAction, Adventure, Fantasy            ', 'PG-13', '8.4'),
 ('Sonic the Hedgehog 2', '\nAction, Adventure, Comedy            ', 'PG', '6.8'),
 ('The Lost City', '\nAction, Adventure, Comedy            ', 'PG-13', '6.6'),
 ('Dune', '\nAction, Adventure, Drama            ', 'PG-13', '8.1'),
 ('The Survivor', '\nBiography, Drama, Sport            ', 'R', '6.8'),
 ('Morbius', '\nAction, Adventure, Horror            ', 'PG-13', '5.1'),
 ('Metal Lords', '\nComedy, Music            ', 'R', '6.7'),
 ('Downton Abbey: A New Era', '\nDrama, Romance            ', 'PG', '7.6'),
 ('Moonfall', '\nAction, Adventure, Sci-Fi            ', 'PG-13', '5.2'),
 ('CODA', '\nComedy, Drama, Music            ', 'PG-13', '8.0'),
 ('The Outfit', '\nCrime, Drama, Mystery            ', 'R', '7.1'),
 ('The Adam Project', '\nAction, Adventure, Comedy            ', 'PG-13', '6.7'),
 ('Turning Red', '\nAnimation, Adventure, Comedy            ', 'PG', '7.0'),
 ('All the Old Knives', '\nMystery, Thriller            ', 'R', '6.1'),
 ('Sing 2', '\nAnimation, Adventure, Comedy            ', 'PG', '7.4'),
 ('Father Stu', '\nBiography, Drama            ', 'R', '7.0'),
 ("The King's Man", '\nAction, Adventure, Thriller            ', 'R', '6.3'),
 ('No Time to Die', '\nAction, Adventure, Thriller            ', 'PG-13', '7.3'),
 ('The Bubble', '\nComedy            ', 'R', '4.7'),
 ('Encanto', '\nAnimation, Comedy, Family            ', 'PG', '7.2'),
 ('Eternals', '\nAction, Adventure, Fantasy            ', 'PG-13', '6.4'),
 ('Memory', '\nAction, Thriller            ', 'R', '5.6'),
 ('Nightmare Alley', '\nCrime, Drama, Thriller            ', 'R', '7.1'),
 ('Deep Water', '\nDrama, Mystery, Thriller            ', 'R', '5.4'),
 ('Scream', '\nHorror, Mystery, Thriller            ', 'R', '6.4'),
 ('The Lighthouse', '\nDrama, Fantasy, History            ', 'R', '7.4'),
 ('Fresh', '\nComedy, Horror, Thriller            ', 'R', '6.7'),
 ('Venom: Let There Be Carnage', '\nAction, Adventure, Sci-Fi            ', 'PG-13', '6.0'),
 ('The Gentlemen', '\nAction, Comedy, Crime            ', 'R', '7.8'),
 ('Wrath of Man', '\nAction, Crime, Thriller            ', 'R', '7.1'),
 ('House of Gucci', '\nCrime, Drama            ', 'R', '6.6'),
 ('Free Guy', '\nAction, Adventure, Comedy            ', 'PG-13', '7.1'),
 ('Ghostbusters: Afterlife', '\nAdventure, Comedy, Fantasy            ', 'PG-13', '7.1'),
 ("Don't Look Up", '\nComedy, Drama, Sci-Fi            ', 'R', '7.2'),
 ('West Side Story', '\nCrime, Drama, Musical            ', 'PG-13', '7.2'),
 ('Avengers: Endgame', '\nAction, Adventure, Drama            ', 'PG-13', '8.4'),
 ('Licorice Pizza', '\nComedy, Drama, Romance            ', 'R', '7.3'),
 ('Sonic the Hedgehog', '\nAction, Adventure, Comedy            ', 'PG', '6.5'),
 ('The Suicide Squad', '\nAction, Adventure, Comedy            ', 'R', '7.2'),
 ('Snake Eyes', '\nAction, Adventure, Crime            ', 'PG-13', '5.3'),
 ('The Power of the Dog', '\nDrama, Romance, Western            ', 'R', '6.9'),
 ("Zack Snyder's Justice League", '\nAction, Adventure, Fantasy            ', 'R', '8.0'),
 ('The Matrix Resurrections', '\nA

This query selects the total number movies from the list of top 250 movies and sorts them into bins based on genre,  evidently, consumer preference is centered around Action and Drama films

In [75]:
%%sql
SELECT
    COUNT(CASE WHEN genre REGEXP 'Action' THEN 1 ELSE NULL END) AS Action,
    COUNT(CASE WHEN genre REGEXP 'Adventure' THEN 1 ELSE NULL END) AS Adventure,
    COUNT(CASE WHEN genre REGEXP 'Drama' THEN 1 ELSE NULL END) AS Drama,
    COUNT(CASE WHEN genre REGEXP 'Comedy' THEN 1 ELSE NULL END) AS Comedy,
    COUNT(CASE WHEN genre REGEXP 'Romance' THEN 1 ELSE NULL END) AS Romance,
    COUNT(CASE WHEN genre REGEXP 'Mystery' THEN 1 ELSE NULL END) AS Mystery,
    COUNT(CASE WHEN genre REGEXP 'Thriller' THEN 1 ELSE NULL END) AS Thriller,
    COUNT(CASE WHEN genre REGEXP 'Horror' THEN 1 ELSE NULL END) AS Horror,
    COUNT(CASE WHEN genre REGEXP 'Family' THEN 1 ELSE NULL END) AS Family,
    COUNT(CASE WHEN genre REGEXP 'Animation' THEN 1 ELSE NULL END) AS Animation,
    COUNT(CASE WHEN genre REGEXP 'Sports' THEN 1 ELSE NULL END) AS Sports
FROM movie;

 * mysql://admin:***@movie-list.crvcl4hdswes.us-east-1.rds.amazonaws.com/movie_list
1 rows affected.


[(92, 72, 118, 69, 22, 31, 60, 39, 7, 15, 0)]

This query displays the average ratings per genre from the previous query of the top three most popular genres

In [90]:
%%sql
WITH review_scores AS ( 
    SELECT
        CASE WHEN genre REGEXP 'Action' THEN review_score ELSE NULL END AS Action_review_scores,
        CASE WHEN genre REGEXP 'Adventure' THEN review_score ELSE NULL END AS Adventure_review_scores,
        CASE WHEN genre REGEXP 'Drama' THEN review_score ELSE NULL END AS Drama_review_scores
    FROM movie
)
SELECT
    ROUND(AVG(Action_review_scores),2) AS avg_action,
    ROUND(AVG(Adventure_review_scores),2) AS avg_adventure,
    ROUND(AVG(Drama_review_scores),2) AS avg_drama,
    MAX(Action_review_scores) AS max_action,
    MAX(Adventure_review_scores) AS max_adventure,
    MAX(Drama_review_scores) AS max_drama
FROM review_scores;

 * mysql://admin:***@movie-list.crvcl4hdswes.us-east-1.rds.amazonaws.com/movie_list
1 rows affected.


[(6.36, 6.57, 6.78, '8.8', '8.8', '8.4')]

This query displays the films that performed better than the average (around 6.47)

In [91]:
%%sql
SELECT 
    m.title,
    m.genre,
    m.review_score
FROM movie m
WHERE m.review_score > 
    (SELECT ROUND(AVG(review_score),2) 
    FROM movie);

 * mysql://admin:***@movie-list.crvcl4hdswes.us-east-1.rds.amazonaws.com/movie_list
145 rows affected.


[('The Batman', '\nAction, Crime, Drama            ', '8.0'),
 ('The Northman', '\nAction, Adventure, Drama            ', '7.8'),
 ('Everything Everywhere All at Once', '\nAction, Adventure, Comedy            ', '8.8'),
 ('The Unbearable Weight of Massive Talent', '\nAction, Comedy, Crime            ', '7.5'),
 ('X', '\nHorror, Thriller            ', '6.7'),
 ('The Black Phone', '\nHorror            ', '7.4'),
 ('Fantastic Beasts: The Secrets of Dumbledore', '\nAdventure, Family, Fantasy            ', '6.5'),
 ('The Bad Guys', '\nAnimation, Adventure, Comedy            ', '6.9'),
 ('Spider-Man: No Way Home', '\nAction, Adventure, Fantasy            ', '8.4'),
 ('Sonic the Hedgehog 2', '\nAction, Adventure, Comedy            ', '6.8'),
 ('The Lost City', '\nAction, Adventure, Comedy            ', '6.6'),
 ('Dune', '\nAction, Adventure, Drama            ', '8.1'),
 ('The Survivor', '\nBiography, Drama, Sport            ', '6.8'),
 ('Metal Lords', '\nComedy, Music            ', '6.7'),
 ('Downton Abbey: A New Era', '\nDrama, Romance            ', '7.6'),
 ('CODA', '\nComedy, Drama, Music            ', '8.0'),
 ('The Outfit', '\nCrime, Drama, Mystery            ', '7.1'),
 ('The Adam Project', '\nAction, Adventure, Comedy            ', '6.7'),
 ('Turning Red', '\nAnimation, Adventure, Comedy            ', '7.0'),
 ('Sing 2', '\nAnimation, Adventure, Comedy            ', '7.4'),
 ('Father Stu', '\nBiography, Drama            ', '7.0'),
 ('No Time to Die', '\nAction, Adventure, Thriller            ', '7.3'),
 ('Encanto', '\nAnimation, Comedy, Family            ', '7.2'),
 ('Nightmare Alley', '\nCrime, Drama, Thriller            ', '7.1'),
 ('The Lighthouse', '\nDrama, Fantasy, History            ', '7.4'),
 ('Fresh', '\nComedy, Horror, Thriller            ', '6.7'),
 ('The Gentlemen', '\nAction, Comedy, Crime            ', '7.8'),
 ('Wrath of Man', '\nAction, Crime, Thriller            ', '7.1'),
 ('House of Gucci', '\nCrime, Drama            ', '6.6'),
 ('Free Guy', '\nAction, Adventure, Comedy            ', '7.1'),
 ('Ghostbusters: Afterlife', '\nAdventure, Comedy, Fantasy            ', '7.1'),
 ("Don't Look Up", '\nComedy, Drama, Sci-Fi            ', '7.2'),
 ('West Side Story', '\nCrime, Drama, Musical            ', '7.2'),
 ('Avengers: Endgame', '\nAction, Adventure, Drama            ', '8.4'),
 ('Licorice Pizza', '\nComedy, Drama, Romance            ', '7.3'),
 ('Sonic the Hedgehog', '\nAction, Adventure, Comedy            ', '6.5'),
 ('The Suicide Squad', '\nAction, Adventure, Comedy            ', '7.2'),
 ('The Power of the Dog', '\nDrama, Romance, Western            ', '6.9'),
 ("Zack Snyder's Justice League", '\nAction, Adventure, Fantasy            ', '8.0'),
 ('Operation Mincemeat', '\nDrama, War            ', '6.9'),
 ('Joker', '\nCrime, Drama, Thriller            ', '8.4'),
 ('Bodies Bodies Bodies', '\nComedy, Horror, Thriller            ', '6.6'),
 ('Once Upon a Time... In Hollywood', '\nComedy, Drama            ', '7.6'),
 ('Tenet', '\nAction, Adventure, Sci-Fi            ', '7.3'),
 ('Nobody', '\nAction, Crime, Drama            ', '7.4'),
 ('The Lion King', '\nAnimation, Adventure, Drama            ', '6.8'),
 ('Midsommar', '\nDrama, Horror, Mystery            ', '7.1'),
 ('The French Dispatch', '\nComedy, Drama, Romance            ', '7.2'),
 ('King Richard', '\nBiography, Drama, Sport            ', '7.5'),
 ('The Last Duel', '\nAction, Drama, History            ', '7.4'),
 ('Dog', '\nComedy, Drama            ', '6.5'),
 ('The Night House', '\nHorror, Mystery, Thriller            ', '6.5'),
 ('Shang-Chi and the Legend of the Ten Rings', '\nAction, Adventure, Fantasy            ', '7.4'),
 ('Jungle Cruise', '\nAction, Adventure, Comedy            ', '6.6'),
 ('Knives Out', '\nComedy, Crime, Drama            ', '7.9'),
 ('Pig', '\nDrama, Mystery, Thriller            ', '6.9'),
 ('1917', '\nAction, Drama, War            ', '8.2'),
 ('The Devil All the Time', '\nCrime, Drama, Thriller            ', '7.1'),
 ('The Green

In [102]:
%%sql
WITH above_average AS ( 
    SELECT 
        m.title,
        m.genre,
        m.review_score
    FROM movie m
    WHERE m.review_score > 
        (SELECT ROUND(AVG(review_score),2) 
        FROM movie)
)
SELECT
    ROW_NUMBER() OVER ( 
    ORDER BY review_score) AS Ranking,
    title,
    genre
FROM movie;

 * mysql://admin:***@movie-list.crvcl4hdswes.us-east-1.rds.amazonaws.com/movie_list
250 rows affected.


[(1, "Fortress: Sniper's Eye", '\nAction            '),
 (2, 'Fortress', '\nAction            '),
 (3, 'Gasoline Alley', '\nAction, Thriller            '),
 (4, 'Agent Game', '\nAction, Thriller            '),
 (5, 'Unplugging', '\nComedy, Romance            '),
 (6, 'Hostile Territory', '\nDrama, History, Western            '),
 (7, 'Cinderella', '\nComedy, Family, Fantasy            '),
 (8, 'Cheaper by the Dozen', '\nAdventure, Comedy, Family            '),
 (9, 'The Last Thing He Wanted', '\nCrime, Drama, Thriller            '),
 (10, 'Space Jam: A New Legacy', '\nAnimation, Adventure, Comedy            '),
 (11, 'The Ice Age Adventures of Buck Wild', '\nAnimation, Adventure, Comedy            '),
 (12, 'Umma', '\nDrama, Horror, Mystery            '),
 (13, 'After We Fell', '\nDrama, Romance            '),
 (14, 'The Devil You Know', '\nThriller            '),
 (15, 'Blacklight', '\nAction, Thriller            '),
 (16, 'The Bubble', '\nComedy            '),
 (17, 'The Desperate Hour', '\nThriller            '),
 (18, 'Texas Chainsaw Massacre', '\nCrime, Horror, Thriller            '),
 (19, 'Horizon Line', '\nAction, Adventure, Drama            '),
 (20, "Charlie's Angels", '\nAction, Adventure, Comedy            '),
 (21, 'Fantasy Island', '\nFantasy, Horror, Mystery            '),
 (22, 'After We Collided', '\nDrama, Romance            '),
 (23, 'Morbius', '\nAction, Adventure, Horror            '),
 (24, 'Moonfall', '\nAction, Adventure, Sci-Fi            '),
 (25, 'F9: The Fast Saga', '\nAction, Crime, Thriller            '),
 (26, 'After', '\nDrama, Romance            '),
 (27, 'The New Mutants', '\nAction, Horror, Mystery            '),
 (28, 'The 355', '\nAction, Thriller            '),
 (29, 'Line of Duty', '\nAction, Crime, Thriller            '),
 (30, 'The Exorcism of God', '\nDrama, Horror            '),
 (31, 'Snake Eyes', '\nAction, Adventure, Crime            '),
 (32, 'Voyagers', '\nAdventure, Sci-Fi, Thriller            '),
 (33, 'Deep Water', '\nDrama, Mystery, Thriller            '),
 (34, 'The Forever Purge', '\nAction, Horror, Thriller            '),
 (35, 'Infinite', '\nAction, Sci-Fi, Thriller            '),
 (36, 'Wonder Woman 1984', '\nAction, Adventure, Fantasy            '),
 (37, 'Moonshot', '\nComedy, Romance, Sci-Fi            '),
 (38, 'Halloween Kills', '\nHorror, Thriller            '),
 (39, 'Wrong Turn', '\nHorror, Thriller            '),
 (40, 'Last Looks', '\nAction, Crime, Mystery            '),
 (41, 'Men in Black: International', '\nAction, Adventure, Comedy            '),
 (42, 'The Marksman', '\nAdventure, Drama, Thriller            '),
 (43, 'Memory', '\nAction, Thriller            '),
 (44, 'Mulan', '\nAction, Adventure, Drama            '),
 (45, 'Escape Room: Tournament of Champions', '\nAction, Adventure, Horror            '),
 (46, 'Without Remorse', '\nAction, Thriller, War            '),
 (47, 'Chaos Walking', '\nAction, Adventure, Fantasy            '),
 (48, 'Studio 666', '\nComedy, Horror, Music            '),
 (49, 'The Contractor', '\nAction, Thriller            '),
 (50, 'Cry Macho', '\nDrama            '),
 (51, 'The Matrix Resurrections', '\nAction, Sci-Fi            '),
 (52, 'Army of the Dead', '\nAction, Crime, Horror            '),
 (53, 'Windfall', '\nCrime, Drama, Thriller            '),
 (54, 'The Rental', '\nDrama, Horror, Mystery            '),
 (55, 'Reminiscence', '\nMystery, Romance, Sci-Fi            '),
 (56, 'The In Between', '\nDrama, Romance, Sci-Fi            '),
 (57, 'Old', '\nDrama, Horror, Mystery            '),
 (58, 'Antlers', '\nDrama, Horror, Mystery            '),
 (59, 'Bill & Ted Face the Music', '\nAdventure, Comedy, Music            '),
 (60, 'Clifford the Big Red Dog', '\nAdventure, Comedy, Family            '),
 (61, 'Candyman', '\nHorror, Thriller            '),
 (62, 'Rumble', '\nAnimation, Comedy, Family            '),
 (63, 'Hotel Transylvania: Transformania', '\nAnimation, Adventure, Comedy            '),
 (64, 'The Voyeurs', '